lib/angstrom.mli

```diff
@@ -426,6 +426,42 @@ val lift4 : ('a -> 'b -> 'c -> 'd -> 'e) -> 'a t -> 'b t -> 'c t -> 'd t -> 'e t
     Even with the partial application, it will be more efficient than the
     applicative implementation. *)
 
+val map2 : 'a t -> 'b t -> f:('a -> 'b -> 'c) -> 'c t
+val map3 : 'a t -> 'b t -> 'c t -> f:('a -> 'b -> 'c -> 'd) -> 'd t
+val map4 : 'a t -> 'b t -> 'c t -> 'd t -> f:('a -> 'b -> 'c -> 'd -> 'e) -> 'e t
+(** The [mapn] family of functions are just like [liftn], with a slightly
+    different interface. *)
+
+val bind : 'a t -> f:('a -> 'b t) -> 'b t
+(** [bind] is a prefix version of [>>=] *)
+
+val map : 'a t -> f:('a -> 'b) -> 'b t
+(** [map] is a prefix version of [>>|] *)
+
+val both : 'a t -> 'b t -> ('a * 'b) t
+(** [both p q] runs [p] followed by [q] and returns both results in a tuple *)
```

Comment:
Is it necessary for all the preceding to be exposed in order for ppx_let to work? Or do you just need the `Let_syntax` module below?

Reply:
Nope, not necessary.  I figured with those functions implemented we might as well expose them, but I wouldn't object to shrinking the interface a bit.

Reply:
I just reorganized them.